### PR TITLE
(CDAP-13234) Refactor Guice modules used in various program TwillRunnable

### PR DESCRIPTION
**Pull request by chtyim**
- Isolates guice module used by TwillRunnable to provide finer gain control
  - Rename DistributedProgramRunnableModule to DistributedProgramContainerModule
    - Turn this class into actual Guice Module class
    - Use builder pattern to build this module class

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    modules.add(new IOModule());
+    modules.add(new ZKClientModule());
+    modules.add(new MetricsClientRuntimeModule().getDistributedModules());
+    modules.add(new MessagingClientModule());
+    modules.add(new LocationRuntimeModule().getDistributedModules());
+    modules.add(new DiscoveryRuntimeModule().getDistributedModules());
+    modules.add(new DataFabricModules(txClientId).getDistributedModules());
+    modules.add(new DataSetsModules().getDistributedModules());
+    modules.add(new AuditModule().getDistributedModules());
+    modules.add(new NamespaceClientRuntimeModule().getDistributedModules());
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java:142]: shouldn't we have different bindings for `ISOLATED` ClusterMode:
```
namespace client
datafabric
dataset
audit
```

Comment (chtyim) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java:142]: This PR is for refactoring, not focusing on what’s needed in the isolated mode. The next PR will have that

Comment (anwar6953) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java:142]: ok, then for the scope of this PR - LGTM.

**Review by anwar6953 — APPROVED** [review]

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramContainerModule.java:142]: I guess we actually don’t need the mode in this module. Could push these down to the AbstractProgramTwillRunnable level

**Review by anwar6953 — APPROVED** [review]

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6370-26

**Comment by chtyim** [comment]
Build passed

**Pull request merged**